Topics/Development_Process/GraphQL_VS_REST.md

```diff
@@ -0,0 +1,94 @@
+# GraphQL and REST APIs
+
+## Table of Contents
+1. [Overview & Motivation](#overview--motivation)
+    - [What is an API?](#what-is-an-api)
+    - [Why do we need APIs?](#why-do-we-need-apis)
+    - [Designing APIs](#designing-apis)
+2. [REST Overview](#rest-overview)
+3. [GraphQL Overview](#graphql-overview)
+4. [When does REST make sense to use?](#when-does-rest-make-sense-to-use)
+5. [When does GraphQL make sense to use?](#when-does-graphql-make-sense-to-use)
+6. [Quick Reference](#quick-reference-for-common-uses)
+7. [Summary](#summary)
+8. [Additional Readings](#additional-reading)
+
+## Overview & Motivation
+### What is an API?
+An API is an **application programming interface**, or in simpler words, a way for two pieces of software to communicate with one another.
+
+On this page, we will refer to things called `endpoints`. These are basically access ways for our API - you use these endpoints to communicate with the software.
+
+### Why do we need APIs?
+Without APIs, there would be no way for software to communicate between one another. In fact, even within the same project, different components may have APIs between them to facilitate communication. Without APIs, software would never be able to interact with anything. Even something as simple as loading Google's webpage deals with many dozens of APIs without you knowing! 
+
+### Designing APIs
+In this article, we will specifically discuss APIs that deal with sending requests over the internet, which is a very special case of APIs. We will cover two special commonly used types of APIs:
+1. REST
+2. GraphQL 
+
+On this page, you will learn about the differences between the two, and hopefully be able to make a confident decision on which one you need to use, based on your project requirements. 
+
+## REST Overview
+A REST API is any API that follows the REST architecture. The implementation of it can be handled in many pre-existing libraries, so this page will focus on the important details:
+1. In REST, you have to specify the `type` of the request you are sending when you call an API endpoint.
+    + Some common examples include `GET`, `POST`, `PUT`, and `DELETE`. Others exist, but these are the ones used most often.
+2. REST is not specific; when you call an endpoint, you will get *all* of the data that endpoint can give, whether you like it or not.

```

Comment:
I'm a little confused about the "you will get all of the data that endpoint can give", could you clarify what this means?

Reply:
I mean that there is no way to specify exactly what data you want; for example, if you are trying to get some data for a specific User through REST, when you call the GET User endpoint, you can't specify that you just want their first name. You have to GET their entire User object and then parse the response to get their first name. I will update this to be more clear!

Reply:
I made the change now, please let me know if you think that clarifies what your question was :)

Reply:
Oh alright, that makes total sense. I think it was just the wording that threw me off. Your changes clarify this, thanks!

Reply:
Awesome! Thank you so much. Do you think there is any ambiguity in any other part of this section?

Reply:
Nope, I think the rest of this section is very clear and gives a good overview of REST. Nice job!